ProtocolHandler.pm

```diff
@@ -1,63 +1,130 @@
 package Plugins::RadioParadise::ProtocolHandler;
 
 use strict;
-# XXX - https
-use base qw(Slim::Player::Protocols::HTTP);
+use base qw(IO::Handle);
 
 use JSON::XS::VersionOneAndTwo;
 use Tie::Cache::LRU;
 use POSIX qw(ceil);
-
-use Slim::Networking::SimpleAsyncHTTP;
+use List::Util qw(min max);
+use AnyEvent::DNS;
 
 use Slim::Utils::Log;
 use Slim::Utils::Prefs;
 use Slim::Utils::Timers;
+use Slim::Utils::Errno;
 
 use Plugins::RadioParadise::Stations;
 
 use constant BASE_URL => 'https://api.radioparadise.com/api/get_block?bitrate=4&chan=%s&info=true%s';
+use constant MAX_ERRORS	=> 5;
 
-# skip very short segments, like eg. some announcements, they seem to cause timing or buffering issues
-use constant MIN_EVENT_LENGTH => 15;
+use constant IDLE 		=> 1;
+use constant CONNECTING => 2;
+use constant CONNECTED 	=> 3;
 
 my $prefs = preferences('plugin.radioparadise');
-tie my %blockData, 'Tie::Cache::LRU', 16;
-
 my $log = logger('plugin.radioparadise');
 
-# To support remote streaming (synced players, slimp3/SB1), we need to subclass Protocols::HTTP
+tie my %blockData, 'Tie::Cache::LRU', 16;
+
 sub new {
 	my $class  = shift;
 	my $args   = shift;
-
-	my $client = $args->{client};
-
-	my $song      = $args->{'song'};
-	my $streamUrl = $song->streamUrl() || return;
-
-	my $sock = $class->SUPER::new( {
-		url     => $streamUrl,
-		song    => $args->{'song'},
-		client  => $client,
-		bitrate => 850_000,
-	} ) || return;
-
-	${*$sock}{contentType} = 'audio/flac';
-
+	my $song = $args->{'song'};
+
+	my $streamUrl = $song->streamUrl || return;
+	my $sock = $class->SUPER::new;
+
+	${*$sock}{'vars'}   = { 	       		# variables which hold state for this instance:
+			'url'			=> $streamUrl,	# Url to grab	
```

Comment:
Could you please clean up whitespace? Tabs in the front, spaces in between code and comments?...